### PR TITLE
Update to manifest v3

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description
V2 of the web extensions manifest is being deprecated by Chrome [[1](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/)]. [V3 of the manifest](https://developer.chrome.com/docs/extensions/mv3/intro/) is in Developer preview right now in Firefox, but already Chrome is warning about it stopping support for it soon.

I have followed the migration guide offered by Google - [Manifest V3 migration checklist](https://developer.chrome.com/docs/extensions/mv3/mv3-migration-checklist/).

## How to test

This should be as easy as importing the extension into Chrome and checking the extension console for errors.

I would wait with merging this until Firefox ships it for everyone.

### CODE REVIEW

**Review by figuemon — APPROVED** [review]

**Review by Siddhant-K-code — COMMENTED** [review]
Doesn't seems to be working 😞 

image

**Comment by filiptronicek** [comment]
Nooooooo 😭😭😭😭

Will take another look, thanks for posting the error, @Siddhant-K-code! 🙏🧡

**Comment by Siddhant-K-code** [comment]
I am on it 🏃🏽.
Will send out a fix ASAP🤞🏽

File: src/gitpodify.ts [hunk 1]
```diff
@@ -15,91 +16,91 @@ let observer: MutationObserver | undefined;
...
+      browser.tabs.create(config.gitpodURL);
     }
+    return;
+  }
+
+  //   BREAKING, TODO: Fix this
+  if (injectedByUserClick) {
+    // User clicked on the Gitpod extension icon. We open the Gitpod with this page as context.
+    browser.tabs.create(renderGitpodUrl(config.gitpodURL));
+  }
```
Comment (Siddhant-K-code) [inline src/gitpodify.ts:50]: related to https://github.com/gitpod-io/browser-extension/pull/65#discussion_r1064388992

File: src/injectors/injector.ts [hunk 2]
```diff
@@ -1,115 +1,137 @@
...
-    const currentUrl = window.location.href;
-    window.open(`https://gitpod.io/#${currentUrl}`, inNewTab ? '_blank' : '_self');
-    e.preventDefault();
-    e.stopPropagation();
+  const currentUrl = window.location.href;
+  //   !BREAKING: Popup options would not work from now on
+  browser.tabs.create({ url: `https://gitpod.io/#${currentUrl}` });
+  e.preventDefault();
+  e.stopPropagation();
 }
```
[the remark this answers is not in the document]
Comment (filiptronicek) [reply src/injectors/injector.ts:77]: Oh yes, re-implementing this with tabs.update() to rewrite the current tab the user is on.

File: src/gitpodify.ts [hunk 3]
```diff
@@ -15,91 +16,97 @@ let observer: MutationObserver | undefined;
...
+  //   BREAKING, TODO: Fix this
+  if (injectedByUserClick) {
+    // User clicked on the Gitpod extension icon. We open the Gitpod with this page as context.
+    try {
+      const currentTab = await browser.tabs.getCurrent();
+      openLink(`${config.gitpodURL}/#${currentTab.url}`);
+    } catch (e) {
+      console.error(e)
+      openLink(config.gitpodURL);
     }
```
Comment (filiptronicek) [inline src/gitpodify.ts:55]: It looks like still the browser action (clicking on the Gitpod logo from the menu bar) only opens gitpod.io, not the repo in Gitpod. I have been trying to debug it, but so far unsuccessful - really not sure what has gone wrong.

@Siddhant-K-code do we have a good way of telling if the `gitpodify.js` is running and if it is, somehow debug it? 👀

Comment (Siddhant-K-code) [reply src/gitpodify.ts:55]: > It looks like still the browser action (clicking on the Gitpod logo from the menu bar) only opens gitpod.io, not the repo in Gitpod. I have been trying to debug it, but so far unsuccessful - really not sure what has gone wrong.

In the same boat ⛵ 😭 

> @Siddhant-K-code do we have a good way of telling if the gitpodify.js is running and if it is, somehow debug it? 👀

I've tried so many workarounds, but I still don't know what we are missing here. Didn't found any way to debug the state of loading `gitpodify.js` 😢

**Comment by filiptronicek** [comment]
I'm getting the feeling something is maybe not right

image

**Comment by Siddhant-K-code** [comment]
Yeah, it seems like DOM is the issue. ([known issue](https://developer.chrome.com/docs/extensions/mv3/known-issues/#offscreen-documents-api)) 😢

**Comment by filiptronicek** [comment]
Superseeded by https://github.com/gitpod-io/browser-extension/pull/81.

**Pull request closed without merging**